### ISSUE TITLE
error installing strumpack in xSDK

### ISSUE THREAD

**Issue by TiffanyAnn** [spack/spack#11202]
I have encountered an error for strumpack when installing xSDK on NERSC Cori.
This error occurs for Intel@18.0 and gcc@7.3.0 compilers (the default versions on Cori).

```spack install xsdk%intel```
```spack install xsdk%gcc```

This is the error I get:
```
  >> 165    $HOME/spack/var/spack/stage/strumpack-3.1.1-rzxid2dujke52kckdqognpnl5q6b2q5u/STRUMPACK-3.1.1/src/BLR/BLRMatrix.hpp(479): error: parallel loops with collapse must be perfectly nested
     166              for (std::size_t j=0; j<colblocks(); j++)
     167                                      ^
```

The section of source code causing the error is:
```
void draw(std::ostream& of, std::size_t roff, std::size_t coff) const {
#if defined(STRUMPACK_USE_OPENMP_TASKLOOP)
#pragma omp taskloop collapse(2) default(shared)
#endif
        for (std::size_t j=0; j<colblocks(); j++)
          for (std::size_t i=0; i<rowblocks(); i++) {
            tile(i, j).draw(of, roff+tileroff(i), coff+tilecoff(j));
          }
      }
```
I noticed in the xsdk package that `strumpack@master` is the version specified to be used for `xsdk@develop` and `strumpack@3.1.1` is used for `xsdk@0.4.0`

The offending code section is from `strumpack@3.1.1`; when I use `strumpack@master` with `xsdk@0.4.0` the error doesn't occur and I am able to successfully install xSDK.

**Comment by TiffanyAnn** [comment]
@balay Can we change which strumpack version is being used for 0.4.0?

**Comment by balay** [comment]
Yeah I saw this error on cori with intel@19 [but not intel@18 which I tested in dec - for xsdk@0.4.0 release.

And @pghysels fixed this in strumpack@develop.

https://github.com/pghysels/STRUMPACK/commits/0b0a5a28132df6cea727361ac3c0a7436dfcf8fb

Perhaps we can grab the fix and apply as a patch to strumpack@3.1.1 in spack.

Ok - I have this change in branch `balay/strumpack-3.1.1-intel-19-fix`. Can you give it a try?

**Comment by TiffanyAnn** [comment]
@balay Yes, I tried it and it worked.
